packages/starlight/user-components/Tabs.astro

```diff
@@ -117,9 +133,13 @@ const { html, panels } = processPanels(panelHtml);
 			});
 		}
 
-		switchTab(newTab: HTMLAnchorElement | null | undefined, index: number) {
+		switchTab(newTab: HTMLAnchorElement | null | undefined, index: number, shouldSync = true) {
 			if (!newTab) return;
 
+			// We store the current position so we can restore it after switching tabs to prevent the page from jumping when
+			// the new tab content is of a different height than the previous tab.
+			const previousTabsOffset = this.getBoundingClientRect().top;
```

Comment:
Might it be worth only calculating this when `shouldSync` is true? Or is that an unnecessary optimisation?

Reply:
Definitely, I don't think this would lead to a performance issue in this case but it's well known that `getBoundingClientRect()` can be a performance bottleneck in more intensive scenarios so we may as well be a good citizen and avoid it when totally unnecessary. Great catch!